### PR TITLE
Introduce `Schema#checkMerge()`

### LINKED ISSUE #1251: Introduce `Schema#checkMerge()`
`Schema#checkMerge()` will check whether two elements can be merged.

See more: https://github.com/ckeditor/ckeditor5-engine/issues/1240.

### PR DESCRIPTION
**Pull request by pomek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Internal: Schema#checkMerge(). Closes #1251.

---

### Additional information

Marked as internal because this whole API is new.

### CODE REVIEW

File: tests/model/schema.js [hunk 1]
```diff
@@ -641,6 +641,48 @@ describe( 'Schema', () => {
...
+		} );
+
+		it( 'returns true if a block can be merged with other block', () => {
+			const listItem = new Element( 'listItem' );
+			const listItemToMerge = new Element( 'listItem', null, [
+				new Text( 'xyz' )
+			] );
+
+			expect( schema.checkMerge( listItem, listItemToMerge ) ).to.equal( true );
+		} );
```
Comment (Reinmar) [inline tests/model/schema.js]: Please add a test where `elementToMerge` has more children and only one of them (not the first one) will not be allowed in the `baseElement`.

Also, another test where `baseElement` already contains disallowed elements, but `elementToMerge` contains only allowed ones. This test will verify stability of the implementation. Such situation is by definition invalid, but current (sufficient) implementation of `checkMerge()` wouldn't reject it, so let's be sure that future implementations won't do that too.

Comment (pomek) [reply tests/model/schema.js]: Missing tests have been added in ddf9e54.

**Comment by Reinmar** [comment]
I realised that `Writer#merge()` accepts a position, so it'd be good if `checkMerge()` also accepted a position (in addition to two elements).

**Comment by pomek** [comment]
@Reinmar, done. Can you check again?

**Comment by Reinmar** [comment]
One thing only – "before/after **the** merge position". I'll fix it on master.

**Pull request merged**